File: 基础（循环、判断、等待）演示.js
/**
 * 小瓶RPA演示demo，具体api请查看*流程开发文档*
 * 官网：https://rpa.pbottle.com/
 * 流程开发文档：https://gitee.com/pbottle/pbottle-rpa/wikis/pages
 */

const pbottleRPA = require('./pbottleRPA')


pbottleRPA.文字转语音('等待3秒开始')
pbottleRPA.日志输出('等待3秒开始');
pbottleRPA.等待(3)


//for 循环 重复操作 10 次 打开网页
for (let index = 0; index < 10; index++) {
    let 计数器 = index+1
    pbottleRPA.日志输出('第' + 计数器 + '次操作'); //输出到日志，文件日志永久保存
    pbottleRPA.打开网址('https://www.baidu.com/s?wd='+计数器) //操作：用默认浏览器打开网页
    pbottleRPA.等待(0.5)
}



//判断
let 随机数 = Math.random()
if (随机数 < 0.5) {        //用 if 判断数值是否小于 0.5
    pbottleRPA.日志输出('小于 0.5。', 随机数)
    pbottleRPA.文字转语音('小于 0.5')
} else {
    pbottleRPA.日志输出('大于或等于 0.5。', 随机数)
    pbottleRPA.文字转语音('大于等于 0.5。')
}
pbottleRPA.等待(3)


//语音播报
pbottleRPA.文字转语音('流程结束！~')
pbottleRPA.显示系统消息('小瓶RPA提示','流程结束！~')

File: python示例/基础（循环、判断、等待）演示.py
"""

PBottle RPA Python version (Beta)
https://gitee.com/pbottle/pbottle-rpa
Example

"""

import pbottleRPA  # Import the PBottle RPA module
import random


# for loop – repeat an operation 10 times, opening a web page each time
for index in range(10):
    number = index + 1
    print("Operation #" + str(number))
    # Output to log, permanently saved in the log file
    pbottleRPA.openURL(
        "https://www.baidu.com/s?wd=" + str(number)
    )  # Action: open a URL with the default browser
    pbottleRPA.sleep(500)  # Wait 500 milliseconds


# Conditional judgment
number = random.random()  # Generate a random number between 0 and 1 for the condition
if number < 0.5:  # Check if the number is less than 0.5
    print("Less than 0.5.", number)
    pbottleRPA.tts("Less than 0.5")
else:
    print("0.5 or greater.", number)
    pbottleRPA.tts("0.5 or greater")


pbottleRPA.wait(1)

# Text-to-speech
pbottleRPA.tts("Flow completed! ~")
pbottleRPA.showMsg("PBottle RPA Tip", "Flow completed! ~")
